lib/polymorphic_integer_type/extensions.rb

```diff
@@ -97,7 +101,28 @@ def has_one(name, scope = nil, **options)
         end
 
         remove_type_and_establish_mapping(name, options, scope)
-        super(name, options.delete(:scope), **options)
+        super(name, options.delete(:scope), **options.except(:foreign_integer_type, :integer_type)).tap do |_|
+          remove_integer_type_and_set_attributes_and_extension(options, ActiveRecord::Reflection::HasOneReflection, reflections[name.to_s])
+        end
+      end
+
+      def remove_integer_type_and_set_attributes_and_extension(options, klass, reflection)
+        foreign_integer_type = options.delete :foreign_integer_type
+        integer_type = options.delete :integer_type
+        is_polymorphic_integer = foreign_integer_type && integer_type
+
+        if is_polymorphic_integer
+          klass.attr_accessor(:foreign_integer_type)
+          klass.attr_accessor(:integer_type)
+          reflection.foreign_integer_type = foreign_integer_type
+          reflection.integer_type = integer_type
+
+          if Gem::Version.new(ActiveRecord::VERSION::STRING) < Gem::Version.new("6.1")
+            ActiveRecord::Associations::Association.prepend(ActiveRecord::Associations::PolymorphicForeignAssociationExtension)
```

Comment:
Do we need to add these modules to the repo?  (Also, not a huge fan of adding them to the activerecord namespaxce when they are ours - they are ours, right?)

Reply:
they are, I follow the file https://github.com/clio/polymorphic_integer_type/blob/01be37705615a5df571b25e1f0e33ed6567680cd/lib/polymorphic_integer_type/belongs_to_polymorphic_association_extension.rb#L3

Reply:
Hmm... I see the pattern we already have, but I do think it'd be more understandable if we named our module after the gem not after active record.  If someone else with ruby knowledge wants to weigh in, happy with another tiebreaker.

Reply:
I am bit back and forth, since adding to the activeRecord namespaces can also explicit show the scope of this module for, but I also get your point.